pkg/kubelet/pod/pod_manager.go

```diff
@@ -153,6 +158,9 @@ func (pm *basicManager) UpdatePod(pod *v1.Pod) {
 // lock.
 func (pm *basicManager) updatePodsInternal(pods ...*v1.Pod) {
 	for _, pod := range pods {
+		if pm.secretManager != nil {
```

Comment:
I run some larger kubemark tests and it didn't seem bad.
That said - I can promise that I will do this optimization (we should discuss how exactly you see it), within say next 2 weeks or so (100% sure that until 1.6 code freeze), but I would really prefer to avoid doing this in this PR and finally have this one in.